libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -791,7 +785,10 @@ void BaseDaemon::initialize(Application& self)
 	signal_listener.reset(new SignalListener(*this));
 	signal_listener_thread.start(*signal_listener);
 
-	graphite_writer.reset(new GraphiteWriter("graphite"));
+	for (const auto & key : DB::config_keys_multi(config(), "", "graphite"))
+	{
+		graphite_writers.emplace(std::make_pair(key, new GraphiteWriter(key)));
```

Comment:
Please avoid operator new, use make_unique instead.
You could construct std::pair inplace, no need for std::make_pair.